### PR TITLE
fix: DIA-1489: Multiskill prompt autorefinement fixes

### PR DESCRIPTION
**Pull request by robot-ci-heartex**
Hi @niklub!

This PR was automaticaly generated via Follow Merge. 
Please ensure that all linked upstream Pull Requests are merged before proceeding with this one.

- https://github.com/HumanSignal/label-studio-sdk/pull/345 linked by [Workflow Run](https://github.com/HumanSignal/Adala/actions/runs/11560414476)

### CODE REVIEW

File: adala/agents/base.py [hunk 1]
```diff
@@ -429,12 +431,16 @@ async def arefine_skill(
             predictions = None
         else:
             inputs = InternalDataFrame.from_records(data or [])
-            predictions = await self.skills.aapply(inputs, runtime=runtime)
-
+            if reapply:
+                predictions = await self.skills.aapply(inputs, runtime=runtime)
+            else:
+                predictions = inputs
```
Comment (matt-bernstein) [inline adala/agents/base.py:437]: in the future, it should be possible to automatically detect if predictions are missing and aapply in that case, and get rid of this `reapply` arg, right?

File: adala/skills/_base.py [hunk 2]
```diff
@@ -633,21 +637,19 @@ def _iter_over_chunks(
             )
         else:
             chunks = [input]
+            
+        # define the row preprocessing function
+        def row_preprocessing(row):
+            return partial_str_format(self.input_template, **row, **extra_fields, i=int(row.name) + 1)
```
Comment (matt-bernstein) [inline adala/skills/_base.py:644]: TY, cleaner this way

File: adala/skills/_base.py [hunk 3]
```diff
@@ -633,21 +637,19 @@ def _iter_over_chunks(
...
-                        self.input_template,
-                        **row, **extra_fields, i=int(row.name) + 1
-                    ),
-                    axis=1,
-                )
+                .apply(row_preprocessing, axis=1)
                 .str.cat(sep=self.input_separator)
             )
-
+            chunk.to_csv('chunk.csv', index=False)
```
Comment (matt-bernstein) [inline adala/skills/_base.py:652]: What is this for? debug left in?

File: adala/agents/base.py (hunk 1 above)
Comment (niklub) [reply adala/agents/base.py:437]: actually it doesn't seem clear to me whether predictions must be applied or not. However, the performance/cost implications are clear. So I'd leave this option for now and put off to decide on the best setup later when we will have some benchmarks to understand the quality tradeoff.

File: adala/skills/_base.py (hunk 3 above)
Comment (niklub) [reply adala/skills/_base.py:652]: oops, debuging stuff

Comment (niklub) [reply adala/skills/_base.py]: fixed

**Pull request merged**